service/mapper/fsm_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package mapper
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/optakt/flow-dps/testing/mocks"
+)
+
+func TestNewFSM(t *testing.T) {
+	st := &State{}
+
+	t.Run("nominal case without options", func(t *testing.T) {
+		t.Parallel()
+
+		f := NewFSM(st)
+
+		assert.NotNil(t, f)
+		assert.Equal(t, st, f.state)
+	})
+
+	t.Run("nominal case with transition option", func(t *testing.T) {
+		t.Parallel()
+
+		f := NewFSM(st, WithTransition(StatusEmpty, func(_ *State) error { return nil }))
```

Comment:
You don't really need to put the `_` if you don't use the parameter. I prefer to only use for code related to business logic, and drop it for anything testing or such.

Reply:
While I agree it doesn't bring much, it's a habit I have to always do that when I don't use the parameter. It doesn't cost much 🤷 Are you fine with keeping it this way?

Reply:
This is too small a detail to disagree over. However, when you put a `_` explicitly, I think it implies _"I ignore this explicitly, and I want to attract your attention to that fact"_. This is rarely the case outside of production code.

In this case, every time you put a `_` in the function signature, it attracts my attention to it, because there should be a reason why you thought it's important enough to notify me of this explicitly. But there isn't :D

Reply:
Mh once again it's interesting how different our interpretation is for stuff like that :p For me, it doesn't necessarily mean that I want to attract attention to the fact that it's ignored. It is just ignored explicitly because it is not used.

Actually for me it's more the opposite, if it's not ignored, I assume it is used somewhere, and I am confused when I see that it is not :p

Reply:
What I mean is that you can remove the `_` completely. You don't need to put a name, either.

Like this:

```go
WithTransition(StatusEmpty, func(*State) error { return nil }))
```

Hence why the presence of a `_` is an indication of wanting to specifically point the absence out.